x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/queries/InferenceQueryUtils.java

```diff
@@ -615,6 +522,141 @@ private static InferenceResults validateAndConvertInferenceResults(
         return inferenceResults;
     }
 
+    private static final class LocalInferenceAsyncAction extends QueryRewriteAsyncAction<
```

Comment:
we could technically reuse this same action for `SparseVectorQueryBuilder` - the problem is this is in the inference plugin, whereas `SparseVectroQueryBuilder` is core.
it's all good with me if we track this separately